students/MzKhan/lesson10/mailroom_fp/cli_main_fp.py

```diff
@@ -0,0 +1,155 @@
+        donor_name = input(input_msg)
+        try:
+            if not "".join(donor_name.split()).isalpha():
+                raise Exception("The name should be all Alphabetic characters:")
+            return donor_name.title()
+        except Exception as e:
+            print("INVALID Name: ",e)
+
+
+def donation_prompt(name):
+    """
+    prompt the user for the donation amount for the given donor.
+    :parm 1: name - required parameter.
+    :return: donation_amount in decimal form.
+    """
+    input_msg = "Please enter the donation amount for {}: "
+    donation_amount = input(input_msg.format(name))
+    while True:
+        try:
+            return float(donation_amount)
+        except ValueError:
+            donation_amount=input("INVALID Amount: "+input_msg.format(name))
+
+
+def display_list(d):
+    """Print the donor's list to the user."""
+    print("\nOur generous donors: \n")
+    for donor_name in iter(d.donors):
+        print(donor_name)
+    print("\n")
+
+
+def quit():
+    """Exit out from the user interaction."""
+    print("Thank you. Have a nice day:")
+    sys.exit()
+
+
+def thank_you(d):
+    """
+    This method was created to utilize the dictionary (options) in the program
+    main flow
+    """
+    input_msg = """Please enter the donor's full name OR
+    \rtype 'list' for the existing donors: """
+    name = name_prompt(input_msg)
+    while name.lower() == "list":
+        display_list(d)
+        name = name_prompt(input_msg)
+    donation = donation_prompt(name)
+    donor = Donor(name, donation)
+    print(donor.thank_you())
+    d.add_donation(name, donation)
+
+
+def print_report(d):
+    """Print the donors' report in the formatted tabular form."""
+    report_data = d.get_report_data()
+    title = "{:24} | {:12} | {:10} | {:20}"
+    dashes=67*('-');print(dashes)
+    print(title.format('Donor Name','Total Given','Num Gifts','Average Gift'))
+    strf_format = "{:24} ${:12.2f}   {:^10d}  ${:12.2f}"
+    print(dashes)
+    for donor in report_data:
+        print(strf_format.format(*donor))
+    print(dashes)
+
+
+def send_letter_everyone(d):
+    """This method creates the letter for each donor in the donor's database"""
+    print("Letters have been sent to all the donors!!!")
+    d.send_letter_everyone()
+
+
+def match_donations(d):
+    print()
+    print("The total amount donated = $ {:.2f}".format(d.total_amount_donated))
+    input_msg = "Please enter the donor's name: "
+    while True:
+        print("Please choose one of the following options:")
+        print("1: Match to total amount donated")
+        print("2: Custom donation")
+        print("3: Leave the menu ")
+        try:
+            user_response = int(input("Option: "))
+            if user_response not in [1,2,3]:
+                print("***INVALID option selected: Try Again!***")
+                continue
+            if user_response == 3:break
+            name = name_prompt(input_msg) ; name = name.title()
```

Comment:
it's a python style violation to use single line multi-statements (both lines 117 and 118)